src/Features/Core/Portable/Workspace/BackgroundCompiler.cs

```diff
@@ -113,8 +113,16 @@ private void Rebuild()
                 return;
 
             using var source = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken.Value, disposalToken);
-            await AddCompilationsForVisibleDocumentsAsync(
-                workspace.CurrentSolution, compilations, source.Token).ConfigureAwait(false);
+            try
+            {
+                await AddCompilationsForVisibleDocumentsAsync(
+                    workspace.CurrentSolution, compilations, source.Token).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
```

Comment:
This change isn't strictly necessary, but helps to highlight the fact that this method has a by-design support for cancelling an in-progress calculation separately from the cancellation token provided by the work queue.

```suggestion
            catch (OperationCanceledException) when (!disposalToken.IsCancellationRequested)
```